wadas/domain/ftps_server.py

```diff
@@ -87,16 +87,25 @@ def on_file_received(self, file):
 
         # check if the received file match one of the allowed extensions
         # (the check relies on an inspection of the file content)
-        if (ftype := filetype.guess(file)) and f".{ftype.extension}" in self.ALLOWED_EXTS:
-            media_queue.put(
-                {
-                    "media_path": file,
-                    "media_id": pathlib.PurePath(file).parent.name,
-                    "camera_id": self.username,
-                }
-            )
+        ftype = filetype.guess(file)
+
+        if ftype:
+            file_ext = ftype.extension
+            logger.debug("Extension of received file: %s", file_ext)
+
+            if f".{ftype.extension}" in self.ALLOWED_EXTS:
+                media_queue.put(
+                    {
+                        "media_path": file,
+                        "media_id": pathlib.PurePath(file).parent.name,
+                        "camera_id": self.username,
+                    }
+                )
+            else:
```

Comment:
Maybe you can return here, when the file is ok. And move the os.remove(file_) to the end of the function, so that you can avoid doing two times in the "clean-up" phase.